### PR TITLE
Upcomingevent

### LINKED ISSUE #3: [Design]: Develop the Upcoming Event Card Component
**Proposed Feature**
Create a reusable card component to display details of upcoming events.

**Describe the solution you'd like**
- Create an EventCard component.
- Use props to pass event details (title, date, time).
- Style the component as per the figma design.

**Additional context**
- Note as a part of this issue you just have to create an **Event card component** i.e. the below component from figma design. Next we will reuse this same event card component for different event by passing there name, date, time, etc to this component.

image

- Your react component might look like below 
   ```
    import React from 'react'
    
    const UpcomingEventCard= ({name, description, date, time, venue, teamsize}) => {
      return (
        <div>
          
        </div>
      )
    }

  export default UpcomingEventCard
   ```

- **Note**: For now, nothing needs to be happening when register is clicked

- You can use **lucide-react** for getting the preferred icons(if any): [https://lucide.dev/icons/]()

### References/Useful links:

- [https://www.freecodecamp.org/news/how-to-build-reusable-react-components/]()

### PR DESCRIPTION
**Pull request by Fenil-1**
Resolves #3 

Added UpcomingEventCard component using props .. 
Register button does not do anything rn as mentioned ..

### CODE REVIEW

**Comment by Saumya40-codes** [comment]
LGTM 🚀
looks perfect



Just few suggestions (please review the below comments)

**Comment by Saumya40-codes** [comment]
The codebase you were/are currently working on might be few commits behind the current codebase which is causing the merge conflict

image


when you do/did ``` git checkout upcomingevent ``` it takes the state of code during that time and creates a `kind of` new copy but it doesnt remain in sync with whatever is updating in the main codebase.

A merge conflict happens when changes in your branch and the main branch affect the same parts of the file(s). Git doesn't know which changes to keep.

Now, for this you can do one of the following option:

1. As you saw in image (or above merge conflict message), you can completely replace `src/App.jsx` and `src/components/Navbar.css` with the current repo's latest change and commit/push again here . (**imo this will be easy way as in your current changes, we wont strictly need the updates that are being made in those two files**)

[check here](https://github.com/Programming-Club-Ahmedabad-University/WOC-2024)



---


2. You can go to cloned repository and make sure your branch is main and click on `Sync Fork` and then `update branch` now your main branch is up-to-date with that of our codebase.

in your local setup, run following commands.

```
git checkout main
```

```
git pull
```

``` 
git checkout upcomingevent
```

```
git merge main
```

now you'll see that merge conflict has occurred message. Now if your using vs code, go to file where merge conflict has occured, and click `open in merge editor`  kind option there. Select `Accept Incoming Changes` this will prioritize the main (latest branch) change in both the file. Once you do this for both file and all conflicts.

Run normal, `git add .`, `git commit` and `git push` commands again.

**Comment by Saumya40-codes** [comment]
Imo it will be simple and right for now that you overwrite `src/App.jsx` and `src/components/Navbar.css` with the current code(the one in this repo) as its ok if what changes you've made in those gets removed as it was going to be changed anyways.

**Comment by Saumya40-codes** [comment]
> Imo it will be simple and right for now that you overwrite `src/App.jsx` and `src/components/Navbar.css` with the current code(the one in this repo) as its ok if what changes you've made in those gets removed as it was going to be changed anyways.

If you didn't understand anything, do comment as I can make those changes from my side as well, but I thought it will be better if you make updates there !

**Comment by Fenil-1** [comment]
> > Imo it will be simple and right for now that you overwrite `src/App.jsx` and `src/components/Navbar.css` with the current code(the one in this repo) as its ok if what changes you've made in those gets removed as it was going to be changed anyways.
> 
> If you didn't understand anything, do comment as I can make those changes from my side as well, but I thought it will be better if you make updates there !

I merged with main .. can you please check it once ?

File: src/components/Navbar.css [hunk 1]
```diff
@@ -1,9 +1,13 @@
 .navbar {
     display: flex;
+<<<<<<< HEAD
```
Comment (Saumya40-codes) [inline src/components/Navbar.css]: remove this section i.e

```
<<<<<<< HEAD
    position: relative;
=======
>>>>>>> main
```

its part of merge conflict

Comment (Fenil-1) [reply src/components/Navbar.css]: done !! now is it perfect ?

Comment (Saumya40-codes) [reply src/components/Navbar.css]: can you push that change? cant see rn here..

same as before
image

Comment (Fenil-1) [reply src/components/Navbar.css]: i pushed it already !!

**Comment by Saumya40-codes** [comment]
Perfect now 🚀

**Comment by Saumya40-codes** [comment]
Thanks @Fenil-1 for this PR

**Pull request merged**